### ISSUE TITLE
Focus capture is not working within overlays

### ISSUE THREAD

**Issue by rschoenbichler** [adobe/react-spectrum#4961]
### Provide a general summary of the issue here

When rendering an overlay/popover, the focus is not captured as mentioned in the documentation

### 🤔 Expected Behavior?

When using the tab key to focus input elements, the focus should stay within the overlay. Once closing the overlay, it should be restored to the element that had focus before the overlay was rendered. This is how it is documented

### 😯 Current Behavior

focus is brought back to the trigger element after tabbing out the last element in the overlay. This causes the overlay to close.

### 🖥️ Steps to Reproduce

https://codesandbox.io/s/fervent-aryabhata-9n279m?file=/src/App.js

### Version

3.18

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

linux

**Comment by snowystinger** [comment]
I think you're just missing a Dialog. See example https://react-spectrum.adobe.com/react-aria/usePopover.html#dialog This is where the Focus Scope is.

**Comment by rschoenbichler** [comment]
But according to [usePopover](https://react-spectrum.adobe.com/react-aria/usePopover.html)

`The Popover component uses an <Overlay> to render its contents in a React Portal at the end of the document body, which ensures it is not clipped by other elements. It also acts as a focus scope, containing focus within the popover and restoring it to the trigger when it unmounts. `

**Comment by snowystinger** [comment]
Ah, you are correct, the docs do say that and I checked, there is a FocusScope in the Overlay component. I think I found the same issue logged a while back. My coworker did a good dive on what is happening.
https://github.com/adobe/react-spectrum/issues/3788#issuecomment-1329777507

At the top of our docs we also state
> Note: usePopover only handles the overlay itself. It should be combined with [useDialog](https://react-spectrum.adobe.com/react-aria/useDialog.html) to create fully accessible popovers. Other overlays such as menus may also be placed in a popover.

I think this is more of a documentation bug at the end of the day. I'm going to close this one as a duplicate.
